packages/docgen/test/get-type-annotation.js

```diff
@@ -415,6 +415,68 @@ describe( 'Type annotations', () => {
 			).toBe( 'string[]' );
 		} );
 
+		it( 'should lean on `getDependants` if `selector` in `createSelector` is missing args', () => {
+			const { tokens } = engine(
+				'test.ts',
+				`/**
+				 * Returns a reference that changes only when the dependencies do.
+				 *
+				 * @param state - stores all the things
+				 */
+				 export const getCountReference = createSelector( () => [], ( state: State ) => [ state.items.length ] );
+				 `
+			);
+
+			expect(
+				getTypeAnnotation(
+					{ tag: 'param', name: 'state' },
+					tokens[ 0 ],
+					0
+				)
+			).toBe( 'State' );
+		} );
+
+		it( 'should ignore `getDependants` if `selector` in `createSelector` has more arguments', () => {
+			const { tokens } = engine(
+				'test.ts',
+				`/**
+				  * Returns a reference that changes only when the dependencies do.
+				  *
+				  * @param state - stores all the things
+				  */
+				 export const getCountReference = createSelector(
+				 	( state: State, id: number ) => [],
+				 	( estate: EState ) => [ estate.items.length ]
+				 );
+				 `
+			);
+
+			expect(
+				getTypeAnnotation(
+					{ tag: 'param', name: 'state' },
+					tokens[ 0 ],
+					0
+				)
+			).toBe( 'State' );
+		} );
+
+		it( 'should ignore `getDependants` if everything in `createSelector` ignores `state`', () => {
+			expect( () =>
+				engine(
+					'test.ts',
+					`/**
+				 * Returns a reference that changes only when the dependencies do.
+				 *
+				 * @param state - stores all the things
+				 */
+				 export const getCountReference = createSelector( () => [], () => [] );
+				 `
+				)
+			).toThrow(
+				/Could not find corresponding parameter token for documented parameter/
+			);
```

Comment:
This is kind of an unfortunate error message. Does the type system handle this case as well so that the user would at least get a helpful message about it? It seems quite unlikely to happen in practice at least.

Reply:
unfortunately I think we don't have a straightforward way of asserting, "this function should have no `@param` tags." when I looked we always start with the `@param` tags and then ask for meta and type information about that. if you can think of a better way I'm open to it, as I really don't like encoding error strings in tests or code. the reason I did so here is to:
 - avoid passing the test when some other failure causes it to break
 - provide some explanation to what we are trying to assert

Reply:
> so that the user would at least get a helpful message about it? It seems quite unlikely to happen in practice at least.

Yeah I think it shows this message on the command line when generating the docs, but it's felt like a common thing in these TypeScript PRs. I see it anywhere we have wrapped functions or values because 

```ts
/**
 * These params don't appear in the parse of the code because
 * whether or not they stick around in `add` is determined at
 * runtime when evaluating `memo()`
 * 
 * @param a
 * @param b
 */
const add = memo( ( a, b ) => a + b );
```

Reply:
Yeah, I think those annotations are wrong, aren't they? I would have annotated that:

```
* @type {(a: unknown, b: unknown) => unknown}
```

:shrug: Unless a const declaration is assigned a function expression, I've always thought that the `@type` tag was more appropriate.

Reply:
Wrong? I don't know. I'd say this is a mismatch between my model of JSDoc views the world and how TypeScript views the world. You're absolutely right. This code is a variable declaration. On the other hand that's a pedantic distinction from the practical reality that it's only and ever serving as a function like any other function declaration.

Practically is it harder to describe the params when using `@type`? Maybe not if we can move to inline parameter descriptions as @mirka recently discussed with React `Props` types. Otherwise it reads more complicated to me, though how much more? I don't know.

Regardless, this is a common practice in the codebase and I'm not sure it would be fully right to update all of those definitions. What do you think?